### PR TITLE
chore(sdk): updating the aws-sdk to version 3.577.0

### PR DESCRIPTION
**Pull request by marciocadev**
This PR updates the aws-sdk past version 3.502, which reduces the cold start runtime on Lambdas according to the article  by David Behroozi: [Coldstarts with the AWS Javascript 3.502 SDK](https://speedrun.nobackspacecrew.com/blog/2024/02/01/coldstarts-with-the-aws-javascript-3502-sdk.html)

I did some tests and in a simple code like the one below, I managed to reduce the cold start from an average of 340ms to an average of 280ms (it's not much but every millisecond counts).

```wing
bring cloud;

let counter = new cloud.Counter() as "my-counter";

new cloud.Function(inflight () => {
  counter.inc();
}) as "my-function";
```

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [ ] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

**Comment by staycoolcall911** [comment]
Hey @marciocadev - what's the status of this PR? Once it's ready for review please let me know and move it out of draft, thank you

**Comment by marciocadev** [comment]
Hey @staycoolcall911 

I'm having an issue with the `getSignedUrl` test, the aws-sdk is a bit tricky to mock and spyOn

File: libs/wingsdk/src/target-tf-aws/function.ts [hunk 1]
```diff
@@ -239,7 +239,15 @@ export class Function extends cloud.Function implements IAwsFunction {
...
-    const bundle = createBundle(this.entrypoint);
+    const bundle = createBundle(this.entrypoint, [
+      "@aws-sdk/client-sso",
+      "@aws-sdk/client-sso-oidc",
+      "@aws-sdk/credential-provider-ini",
+      "@aws-sdk/credential-provider-process",
+      "@aws-sdk/credential-provider-sso",
+      "@aws-sdk/credential-provider-web-identity",
+      "@aws-sdk/token-providers",
+    ]);
```
Comment (Chriscbr) [inline libs/wingsdk/src/target-tf-aws/function.ts:250]: Perhaps we can define this as a shared constant in the shared-aws directory

**Comment by marciocadev** [comment]
@staycoolcall911, I need some help here.

Everything is okay, but the signedUrl test isn't working because I can't mock the getSignedUrl method. I created a project to test by copying this method, and the test I'm using in my project works perfectly.

I'm not able to get this to work, could someone help me with this?


Here's the project with the test working

https://github.com/marciocadev/AWSSignedUrlTest

**Comment by staycoolcall911** [comment]
Hey @marciocadev - of course, I'll find someone to help you out.

**Comment by hasanaburayyan** [comment]
@marciocadev my bad, I for some reason thought https://github.com/winglang/wing/pull/5429 included this change. But I see it only upgraded the sdk to 3.490.0

**Comment by Chriscbr** [comment]
I tried a few approaches (like mocks and spies) within our repo and also got stuck. After seeing it worked in https://github.com/marciocadev/AWSSignedUrlTest, I decided to instead try removing code from our repo to see what was necessary to get it working.

I'm not quite sure why it fixes it, but if I comment out this line in our vitest config, it seems to fix the test. cc @MarkMcCulloh https://github.com/winglang/wing/blob/845360a7877629185cac62ac9464b21a91c12608/libs/wingsdk/vitest.config.mts#L5

**Comment by Chriscbr** [comment]
@marciocadev @hasanaburayyan I feel for now it's better if we upgrade to the latest AWS SDK version and try to address the test mocking issue another time. I've pushed a commit that skips the relevant tests - let's try and get this across the finish line

File: libs/wingsdk/src/shared-aws/function.ts [hunk 2]
```diff
@@ -7,6 +7,16 @@ import { INFLIGHT_SYMBOL } from "../core/types";
...
 
+export const externalLibraries = [
+  "@aws-sdk/client-sso",
+  "@aws-sdk/client-sso-oidc",
+  "@aws-sdk/credential-provider-ini",
+  "@aws-sdk/credential-provider-process",
+  "@aws-sdk/credential-provider-sso",
+  "@aws-sdk/credential-provider-web-identity",
+  "@aws-sdk/token-providers",
+];
```
Comment (MarkMcCulloh) [inline libs/wingsdk/src/shared-aws/function.ts:18]: Does esbuild's `external` option work with `*`?

```suggestion
export const externalLibraries = ["@aws-sdk/*"];
```

Even if so, maybe there's value in being explicit 🤷

**Review by MarkMcCulloh — APPROVED** [review]

**Comment by Chriscbr** [comment]
Spec tests passed: https://github.com/winglang/wing/actions/runs/9133888270

File: libs/wingsdk/src/shared-aws/function.ts (hunk 2 above)
Comment (Chriscbr) [reply libs/wingsdk/src/shared-aws/function.ts:18]: Either is ok to me

**Pull request merged**